scripts/mod_loader/modapi/pawn.lua

```diff
@@ -205,39 +234,398 @@ BoardPawn.ApplyDamage = function(self, spaceDamage)
+	Tests.AssertSignature{
+		ret = "void",
+		func = "SetTeleporter",
+		params = { self, teleporter },
+		{ "userdata|BoardPawn&", "boolean|bool" },
+		{ "userdata|BoardPawn&" }
+	}
+	
+	if teleporter == nil then
+		teleporter = true
 	end
+	
+	CUtils.SetPawnTeleporter(self, teleporter)
+end
 
-	local loc = mouseTile()
-	local p = loc and Board:GetPawn(loc)
-	if p then
-		return p:GetId() == self:GetId()
+BoardPawn.SetJumper = function(self, jumper)
+	Tests.AssertSignature{
+		ret = "void",
+		func = "SetJumper",
+		params = { self, jumper },
+		{ "userdata|BoardPawn&", "boolean|bool" },
+		{ "userdata|BoardPawn&" }
+	}
+	
+	if jumper == nil then
+		jumper = true
 	end
+	
+	CUtils.SetPawnJumper(self, jumper)
+end
+
+BoardPawn.GetMaxHealth = function(self)
+	Tests.AssertSignature{
+		ret = "int",
+		func = "GetMaxHealth",
+		params = { self },
+		{ "userdata|BoardPawn&" }
+	}
+	
+	return CUtils.GetPawnMaxHealth(self)
+end
+
+BoardPawn.GetBaseMaxHealth = function(self)
+	Tests.AssertSignature{
+		ret = "int",
+		func = "GetBaseMaxHealth",
+		params = { self },
+		{ "userdata|BoardPawn&" }
+	}
+	
+	return CUtils.GetPawnBaseMaxHealth(self)
+end
+
+BoardPawn.SetHealth = function(self, hp)
+	Tests.AssertSignature{
+		ret = "void",
+		func = "SetHealth",
+		params = { self, hp },
+		{ "userdata|BoardPawn&", "number|int" }
+	}
+	
+	local hp_max = self:GetMaxHealth()
+	hp = math.max(0, math.min(hp, hp_max))
+	
+	CUtils.SetPawnHealth(self, hp)
+end
+
+BoardPawn.SetMaxHealth = function(self, hp_max)
+	Tests.AssertSignature{
+		ret = "void",
+		func = "SetMaxHealth",
+		params = { self, hp_max },
+		{ "userdata|BoardPawn&", "number|int" }
+	}
+	
+	CUtils.SetPawnMaxHealth(self, hp_max)
+end
+
+BoardPawn.SetBaseMaxHealth = function(self, hp_max_base)
+	Tests.AssertSignature{
+		ret = "void",
+		func = "SetBaseMaxHealth",
+		params = { self, hp_max_base },
+		{ "userdata|BoardPawn&", "number|int" }
+	}
+	
+	CUtils.SetPawnBaseMaxHealth(self, hp_max_base)
+end
+
+
+BoardPawn.MarkHpLoss = function(self, hp_loss)
+	CUtils.PawnMarkHpLoss(self, hp_loss)
+end
+
+
+BoardPawn.GetWeaponCount = function(self)
+	return CUtils.PawnGetWeaponCount(self)
+end
+
+BoardPawn.GetWeaponName = function(self, index)
+	return CUtils.PawnGetWeaponName(self, index)
+end
+
+BoardPawn.GetWeaponType = function(self, index)
+	return CUtils.PawnGetWeaponType(self, index)
+end
+
+BoardPawn.GetWeaponClass = function(self, index)
+	return CUtils.PawnGetWeaponClass(self, index)
+end
+
+BoardPawn.GetMoveSkill = function(self)
+	return CUtils.PawnGetMoveSkill(self)
+end
+
+--[[BoardPawn.GetRepairSkill = function(self)
+	return CUtils.PawnGetRepairSkill(self)
+end]]
+
+
+BoardPawn.RemoveWeapon = function(self, index)
+	return CUtils.PawnRemoveWeapon(self, index)
+end
+
+BoardPawn.SetMoveSkill = function(self, skill)
+	CUtils.PawnSetMoveSkill(self, skill)
+end
+
+BoardPawn.SetRepairSkill = function(self, skill)
+	CUtils.PawnSetRepairSkill(self, skill)
+end
+
+
+BoardPawn.GetPilotId = function(self)
```

Comment:
Seems like there's a lot of functions related to pilots here. Is there no way to separate them from pawn entity, and make them work with pilots in the player's inventory?

Reply:
Well, we don't have any real `Pilot` object, so it has to come from a pawn at some point. I suppose we could have a fake `Pilot` object that we could get with `Pawn:GetPilot()`, which would hold all pilot functions.
I don't think we should just expect 3 pilots from the main 3 mechs by the way, since that limitation might be lifted in the future.

Reply:
I thought about the same thing regarding a fake `Pilot` object, though in that case the question is, how are the pilot-modifying functions implemented in C++ side?  
If the `Pilot` object has to proxy through a pawn memory address, then it's problematic, since mods could hold on to the `Pilot` object reference, while the player assigns a different pilot to the mech. The `Pilot` object would still issue calls through the pawn, modyfing another pilot's data.

A quick glance at the C++ code seems to suggest there's a separate `PilotData` structure, so it seems like it could be feasible?

The vanilla pawn functions relating to the pilot are few in number, and only *return* data. With the number of new functions added here, as well as the fact that they can be used to modify data, I think it's better to separate them. It also makes it easier to manage in the future, since the `Pilot` object and the new functions can be placed in a `pilot.lua` file.

Reply:
I'll do some research and see if I can separate Pilot from Pawn properly. It does appear pilots exists entirely separate from pawns, so it should be doable. If it works out I think it will be fairly easy to move pilots inbetween mechs, and hopefully create new pilot instances at will as well.